### DISCUSSION TITLE
Issues with simulation

### DISCUSSION THREAD

**Discussion by gitfeldhase** [oemof/DHNx#112] [Q&A] [answered]
Hello everyone,

I am facing some issues with the simulation and I just can not figure it out where my mistake is.
I tried to run the example simulation_example.py with the attached csv_network as a tree structure (see .zip -> "simulation_tree").
Somehow, the simulation gives me not the correct temperatures (minus degrees even though) the inlet temperature of the producer is set to 90. I just can not figure it out which parameter is set wrong and compared it alot with the csv_network folder of the example but I just do not see a difference.

I would appreciate if somebody could take a quick look on the attached network data, maybe I am missing something very obviously..
[tree_example_dhnx.ZIP](https://github.com/oemof/DHNx/files/10230564/tree_example_dhnx.ZIP)

Thanks for the great support here!

**EDIT**
Something that I noticed is, the producers-0 to forks-x pipe is listed very first in the example and in my example it is listed in the last line.
Could this make a difference?
In the results, the temperature of my producer is 90 degree celsius which is correct, all the other temperatures are false (equal or close to ambient temperature.)
**End Edit**

Best regards
Lars

**Comment by gitfeldhase** [comment]
I feel like the naming is wrong somewhere and the environment temperature is set to all forks.

**Comment by gitfeldhase** [comment, marked as answer]
> Hello everyone,
> 
> I am facing some issues with the simulation and I just can not figure it out where my mistake is. I tried to run the example simulation_example.py with the attached csv_network as a tree structure (see .zip -> "simulation_tree"). Somehow, the simulation gives me not the correct temperatures (minus degrees even though) the inlet temperature of the producer is set to 90. I just can not figure it out which parameter is set wrong and compared it alot with the csv_network folder of the example but I just do not see a difference.
> 
> I would appreciate if somebody could take a quick look on the attached network data, maybe I am missing something very obviously.. [tree_example_dhnx.ZIP](https://github.com/oemof/DHNx/files/10230564/tree_example_dhnx.ZIP)
> 
> Thanks for the great support here!
> 
> **EDIT** Something that I noticed is, the producers-0 to forks-x pipe is listed very first in the example and in my example it is listed in the last line. Could this make a difference? In the results, the temperature of my producer is 90 degree celsius which is correct, all the other temperatures are false (equal or close to ambient temperature.) **End Edit**
> 
> Best regards Lars

I checked my files and some had as direction -1.
After switching from and to nodes and changing the direction the simulation worked without bugs and temperatures seem to be fine.

**Reply by jnnr** [reply]
Good to hear! Thanks for checking, @gitfeldhase! I implemented the simulation a while ago but never could test it on larger networks. If you like, feel free to add another example based on your calculations!

Sadly, I have not been working on this topic for a while and will not pick it up in the near future. Therefore, I will not be able to spend large amounts of time on DHNx. I learned that knowing about how much original developers are still involved can help, so I wanted to let you know about this. Feel free to take responsibility when finding bugs or you have ideas for improvements!
